File: nreduce/benchmark/nfib.py
#!/usr/bin/python
import sys

def nfib(n):
    if n <= 1:
        return 1
    return nfib(n-2)+nfib(n-1)

if len(sys.argv) < 2:
    sys.stderr.write("no max value supplied\n")
    sys.exit(-1)

max = sys.argv[1]

for i in range(int(max)+1):
    print "nfib(%d) = %d" % (i,nfib(i))

File: nreduce/benchmark/nfib.pl
#!/usr/bin/perl

sub nfib
{
  my ($n) = @_;
  if ($n <= 1) {
    return 1;
  }
  else {
    return nfib($n-2)+nfib($n-1);
  }
}

my $n = ($#ARGV < 0) ? 24 : $ARGV[0];
for ($i = 0; $i <= $n; $i++) {
  printf("nfib(%d) = %d\n",$i,nfib($i));
}
